src/menu/menu.c

```diff
@@ -1153,6 +1268,13 @@ menu_open_root(struct menu *menu, int x, int y)
 		destroy_pipemenus(menu->server);
 	}
 	close_all_submenus(menu);
+	/*
+	 * dynamic internal client
+	 * similar to pipemenu but no script to execute
+	 */
+	create_client_list_combined_menu(menu->server);
+	create_client_send_to_menu(menu->server);
```

Comment:
Can we make this depend on some `menu->rebuild_on_call` function callback? There is not really a point to reload both when a static menu is opened. E.g. something like this:
```c
struct menu_item {
    ..
    void (*rebuild_on_call)(struct menu *menu);
};

..

if (menu->rebuild_on_call) {
    menu->rebuild_on_call(menu);
}
```

Reply:
The reason I have that there is the client menus didn't update properly if a submenu of something like the root menu.

I'm not sure what to do.

Reply:
We also need to have that check in `menu_process_item_selection()` when opening a sub menu.